### PR TITLE
PLFM-6335: Added basic auth service abstraction

### PR DESCRIPTION
**Pull request by marcomarasca**

### CODE REVIEW

**Comment by marcomarasca** [comment]
Merging this requires https://github.com/Sage-Bionetworks/Synapse-Stack-Builder/pull/339

File: services/repository/src/main/java/org/sagebionetworks/auth/filter/StackConfigKeyAndSecretProvider.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ * Default implementation for a {@link ServiceKeyAndSecretProvider} that grabs
+ * the key/secret pair from the stack configuration
+ * 
+ * @author Marco Marasca
+ *
+ */
+public class StackConfigKeyAndSecretProvider implements ServiceKeyAndSecretProvider {
+
+	private String serviceName;
+	private String serviceKey;
```
Comment (john-hill) [inline services/repository/src/main/java/org/sagebionetworks/auth/filter/StackConfigKeyAndSecretProvider.java:15]: Is there a way we can avoid keeping this in memory?

File: services/repository/src/main/java/org/sagebionetworks/repo/web/OAuthScopeInterceptor.java [hunk 2]
```diff
@@ -61,10 +61,20 @@ public static boolean isAnonymous(HttpServletRequest request) {
...
+		String serviceName = request.getHeader(AuthorizationConstants.SYNAPSE_HEADER_SERVICE_NAME);
+		return serviceName != null;
+	}
 
 	@Override
 	public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler)
 			throws Exception {
+
+		// Service calls do not need to have the scope checked since they are authenticated by the filter itself
+		if (isServiceCall(request)) {
```
Comment (john-hill) [inline services/repository/src/main/java/org/sagebionetworks/repo/web/OAuthScopeInterceptor.java:75]: check with @brucehoff on this one.

File: integration-test/src/test/java/org/sagebionetworks/ITMigrationTest.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+		String migrationUser = stackConfig.getMigrationAdminUsername();
+		String migrationKey = stackConfig.getMigrationAdminAPIKey();
+		
+		adminSynapse.setUsername(migrationUser);
+		adminSynapse.setApiKey(migrationKey);
+		
+		// This should still work
+		assertNotNull(adminSynapse.getMigrationTypes());
+	}
+
```
Comment (john-hill) [inline integration-test/src/test/java/org/sagebionetworks/ITMigrationTest.java:55]: add a test that adds the "magic" and ensure it does not work.

**Review by john-hill — APPROVED** [review]

File: services/repository/src/main/java/org/sagebionetworks/repo/web/OAuthScopeInterceptor.java [hunk 4]
```diff
@@ -61,10 +61,20 @@ public static boolean isAnonymous(HttpServletRequest request) {
 				AuthorizationConstants.BOOTSTRAP_PRINCIPAL.ANONYMOUS_USER.getPrincipalId()
 					.equals(Long.parseLong(userIdRequestParameter));
 	}
+	
+	public static boolean isServiceCall(HttpServletRequest request) {
+		String serviceName = request.getHeader(AuthorizationConstants.SYNAPSE_HEADER_SERVICE_NAME);
```
Comment (brucehoff) [inline services/repository/src/main/java/org/sagebionetworks/repo/web/OAuthScopeInterceptor.java:66]: need to make sure that a client cannot add this header to bypass the scope check

**Pull request merged**